go/oasis-node/cmd/common/metrics/resource.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/oasislabs/oasis-core/go/common/service"
+)
+
+// ResourceCollector is interface for monitoring resources (cpu, mem, disk, net).
+type ResourceCollector interface {
+	// Update updates corresponding resource metrics.
+	Update() error
+}
+
+// resourceService updates and collects number of resources.
+type resourceService struct {
+	service.BaseBackgroundService
+
+	interval           time.Duration
+	resourceCollectors []ResourceCollector
+}
+
+func (brs *resourceService) Start() error {
```

Comment:
Could use just `rs` instead of `brs` (it is also what you use in the constructor).

Reply:
Thanks, this remained when it was still called `baseResourceService`.